### PR TITLE
Feature/global menu

### PR DESCRIPTION
**Pull request by chrislombaard**
Added the global menu in
Styled the links to be more uniform with buttons
Added the React imports again to get the demo build to work
Configured the redux store to accept a menu config json object
Menu object is passed accordingly to the ReactWebChatComponent and its children
Menu is populated properly.

### CODE REVIEW

File: src/components/InputArea/index.js [hunk 1]
```diff
@@ -54,15 +52,21 @@ export const InputArea = ({
...
-            {inputExpected === 'radio' && (
-                <MenuComponent items={buttons} submitHandler={submitHandler} />
-            )}
             {inputExpected === 'checkbox' && (
                 <CheckboxMenuComponent
                     items={buttons}
                     submitHandler={submitHandler}
                 />
             )}
+            {
```
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/components/InputArea/index.js]: Ah yes of course. Removed.

File: src/components/InputArea/index.js [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
-        buttons: latestMessage && latestMessage.buttons
+        buttons: config.menu.buttons
     };
 };
 
 const mapDispatchToProps = dispatch => ({
-    submitHandler: text => {
+    submitHandler: payload => {
         dispatch(
-            messageActions.messageSend({
```
Comment (aleciafb) [inline src/components/InputArea/index.js]: @chrislombaard I assume the payload is an object that includes one or more of the following keys: postback, text, type

File: src/themes/default/components/Avatar/_avatar.scss [hunk 3]
```diff
@@ -2,6 +2,7 @@
     position: absolute;
     top: 0;
     width: $avatar-size;
+    margin-left: -7px;
```
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/themes/default/components/Avatar/_avatar.scss]: The avatar was pushing the chat bubbles too far to the right, so we shifted the avatar slightly to the left.

File: src/themes/default/components/Carousel/_carousel.scss [hunk 4]
```diff
@@ -5,17 +5,17 @@
...
     }
 
     &-item {
         display: inline-block;
         vertical-align: top;
-        padding: {
-            left: $message-spacing-thick;
-            right: $message-spacing-thick;
-        }
+        //TODO: Removed this, because arrows were removed, thus padding for them should be removed.
```
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/themes/default/components/Carousel/_carousel.scss]: Yes, removed :)

File: src/themes/default/components/Input/index.js [hunk 5]
```diff
@@ -20,7 +20,7 @@ const enhance = compose(
  * @param {function(text: string)} param.submitHandler - submit handler function. This will send a message to the server using data supplied as parameter
  * @return {Object} React component
  */
-export const Input = ({ submitHandler, onKeyDown, value, setValue }) => {
+export const Input = ({ submitHandler, onKeyDown, value, setValue, menuIcon }) => {
```
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/themes/default/components/Input/index.js]: Forgot to remove this, removed.

File: src/themes/default/components/Input/index.js [hunk 6]
```diff
@@ -48,7 +48,9 @@ export const Input = ({ submitHandler, onKeyDown, value, setValue }) => {
                     setValue('');
                 }}
             >
-                SEND
+                {/*<div className="AvatarStatus-avatar">*/}
```
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/themes/default/components/Input/index.js]: Yes, removed.

File: src/themes/default/components/Menu/_menu.scss [hunk 7]
```diff
@@ -19,8 +19,8 @@
         width: 60%;
         margin-bottom: 4px;
         margin-left: 50%;
-        transition: .2s ease;
-        transform: translate(-50%, 0); 
+        transition: 0.2s ease;
```
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/themes/default/components/Menu/_menu.scss]: Ah yes of course. Removed

File: src/themes/default/components/Menu/_menu.scss [hunk 8]
```diff
@@ -37,16 +37,23 @@
...
         font-family: $helvetica;
         text-align: center;
         color: $grey-dark;
         width: 100%;
         height: 0;
         border: 0;
         padding: 0;
+        font-weight: 600;
+
+        &-itemLink {
```
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/themes/default/components/Menu/_menu.scss]: Yes, you're right. Changed.

File: src/themes/default/components/Menu/index.js [hunk 9]
```diff
@@ -1,7 +1,8 @@
 // @ts-check
-//import React from 'react';
+import React from 'react';
 //import PropTypes from 'prop-types';
 import { compose, withStateHandlers } from 'recompose';
+import { Button } from "../Button";
```
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/themes/default/components/Menu/index.js]: No, removed.

File: src/components/InputArea/index.js (hunk 2 above)
Comment (chrislombaard) [reply src/components/InputArea/index.js]: Yes it is, that fine? @aleciafb

File: src/themes/default/components/Button/_button.scss [hunk 10]
```diff
@@ -49,6 +49,12 @@
     &--solid {
         background-color: $white;
         font-size: inherit;
+        color: $grey-dark;
+
+        a {
```
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/themes/default/components/Button/_button.scss]: Yes you're right. Removed.

File: src/components/InputArea/index.js (hunk 2 above)
Comment (aleciafb) [reply src/components/InputArea/index.js]: Yes thanks!

File: demo/src/index.js [hunk 11]
```diff
@@ -21,15 +21,32 @@ class Demo extends Component {
                 className="ReactWebChat"
             >
                 <ReactWebChatComponent
-                    url="http://localhost:8080/echo/"
+                    url="https://dev.feersum.io/chat_sockjs"
```
Comment (aleciafb) [inline demo/src/index.js]: @chrislombaard Should we commit this? Ideally I would like to move this out into .env files for dev

File: src/themes/default/components/Avatar/_avatar.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (aleciafb) [reply src/themes/default/components/Avatar/_avatar.scss]: @chrislombaard Can we please add a note to make sure it uses the $rythm instead?

File: demo/src/index.js (hunk 11 above)
Comment (chrislombaard) [reply demo/src/index.js]: I keep forgetting to remove this when I push. Sorry. Will remove

File: src/themes/default/components/Avatar/_avatar.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (chrislombaard) [reply src/themes/default/components/Avatar/_avatar.scss]: Cool, changed.

**Pull request merged**